case_mapping/uco/observable.py

```diff
@@ -426,6 +428,70 @@ def __init__(self, *args: Any, has_changed=None, state=None, **kwargs: Any) -> N
         self._bool_vars(**{"uco-observable:hasChanged": has_changed})
 
 
+class ObservableAction(Action):
+    def __init__(
+        self,
+        *args: Any,
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+        action_count: Optional[int] = None,
+        action_status: Optional[str] = None,
+        end_time: Optional[datetime] = None,
+        environment: Optional[ObjectEntity] = None,
+        error: Optional[ObjectEntity] = None,
+        instrument: Union[None, ObjectEntity, List[ObjectEntity]] = None,
+        location: Union[None, Location, List[Location]] = None,
+        object: Union[None, ObjectEntity, List[ObjectEntity]] = None,
+        participant: Optional[ObjectEntity] = None,
+        performer: Optional[ObjectEntity] = None,
+        result: Union[None, ObjectEntity, List[ObjectEntity]] = None,
+        start_time: Optional[datetime] = None,
+        subaction: Optional[ObjectEntity] = None,
+        **kwargs: Any,
+    ) -> None:
+        """
+        An observable action is a grouping of characteristics unique to something that may be done or performed within the digital domain.
+        """
+        super().__init__()
```

Comment:
Please use the `*args, **kwargs` delegation pattern for `super()` here.  It will let you trim away keyword properties here that are already handled far up in the parent class `ObjectEntity`.
EDIT - Nevermind, see last comment in this thread.

Reply:
E.g. `name`, `description` would be handled by `ObjectEntity`; `participant`, `performer` should be handled by `Action`.

Reply:
Checking back at the ontology, I think this entire `__init__` method is completely redundant.  `observable:ObservableAction` has no further properties.

Reply:
I pushed an update that ports your additions to the `ObservableAction` class up to the superclass.  This `__init__` method remains, but it's reduced to assigning `@type`---which, from the ontology, is it all it needs to do.